src/seeder.c

```diff
@@ -1135,55 +1134,61 @@ void seeder_high_vol_random(int Nx, int Ny, int Nz, double bias, int times, real
+  real *y_new;
+  y_new = malloc(nparts * sizeof(real));
+  real *z_new;
+  z_new = malloc(nparts * sizeof(real));
+  
+  double d_x = 0.0;
+  double d_y = 0.0;
+  double d_z = 0.0;
+  double d_min = 100 * a;
+  double d_pair= 100 * a;;
+  srand(time(NULL));
+  for(int t = 0; t < times; t++)
   {
-  for(int k = 0; k < Nz; k++)
-   {
-    for (int j = 0; j < Ny; j++)
-     {
-      for (int i = 0; i < Nx; i++)
-       { 
-        d_min = 100*a;
-        x_new = parts[i + j*Nx + k*(Nx*Ny)].x + bias*a*(-1 + 2*((double) rand() /(double) (RAND_MAX))); //make the value to become -1 to 1
-        y_new = parts[i + j*Nx + k*(Nx*Ny)].y + bias*a*(-1 + 2*((double) rand() /(double) (RAND_MAX)));
-        z_new = parts[i + j*Nx + k*(Nx*Ny)].z + bias*a*(-1 + 2*((double) rand() /(double) (RAND_MAX)));
-        if (x_new > Dom.xs  && x_new < Dom.xe && y_new > Dom.ys && y_new < Dom.ye && z_new > Dom.zs && z_new < Dom.ze)
-        { 
-          for (int n = 0; n < Nz; n++)
-          {
-            for (int m = 0; m < Ny; m++)
-            {
-              for (int l = 0; l < Nx; l++)
-              {
-              if(i == l && j == m && k==n)d_pair = 100*a; //if it calculates the distance to itself
-              else{
-                d_pair = (x_new - parts[l + m*Nx + n*(Nx*Ny)].x)*(x_new - parts[l + m*Nx + n*(Nx*Ny)].x) 
-                + (y_new - parts[l + m*Nx + n*(Nx*Ny)].y)*(y_new - parts[l + m*Nx + n*(Nx*Ny)].y)
-                + (z_new - parts[l + m*Nx + n*(Nx*Ny)].z)*(z_new - parts[l + m*Nx + n*(Nx*Ny)].z);
-                d_pair = sqrt(d_pair);
-              } 
-              if (d_pair < d_min){ 
-                d_min = d_pair; //find the minimum distance between particle pairs
-              } 
-              }
-            }
-          }
-          if (d_min > 2*a){   //accept the perturbation if particles don't interact with each other
-            parts[i + j*Nx + k*(Nx*Ny)].x = x_new;
-            parts[i + j*Nx + k*(Nx*Ny)].y = y_new;
-            parts[i + j*Nx + k*(Nx*Ny)].z = z_new;
-          } 
-        }
+    for(int i = 0; i < nparts; i++)
+    {
+      x_new[i] = parts[i].x + bias*(-1 + 2*((double) rand() /(double) (RAND_MAX)))*a*2.0;
+      y_new[i] = parts[i].y + bias*(-1 + 2*((double) rand() /(double) (RAND_MAX)))*a*2.0;
+      z_new[i] = parts[i].z + bias*(-1 + 2*((double) rand() /(double) (RAND_MAX)))*a*2.0;
+      if(x_new[i] > Dom.xe) x_new[i] = x_new[i] - Dom.xl;
+      if(x_new[i] < Dom.xs) x_new[i] = x_new[i] + Dom.xl;
+      if(y_new[i] > Dom.ye) y_new[i] = y_new[i] - Dom.yl;
+      if(y_new[i] < Dom.ys) y_new[i] = y_new[i] + Dom.yl;
+      if(z_new[i] > Dom.ze) z_new[i] = z_new[i] - Dom.zl;
+      if(z_new[i] < Dom.zs) z_new[i] = z_new[i] + Dom.zl; 
+      d_min = 100 * a;
+      for(int j = 0; j < nparts; j++)
+      {
+        d_x = x_new[i] - parts[j].x;
+        d_y = y_new[i] - parts[j].y;
+        d_z = z_new[i] - parts[j].z;
+        if(fabs(d_x) > Dom.xl/2.0) d_x = Dom.xl - fabs(d_x);
```

Comment:
check for various boundary conditions.
If, for `DIRICHLET` boundary conditions:
* `xl = 10*r`
* `parts[i].x = r`
* `parts[j].x = xl - r = 9*r`

Then
* `d_x = 8*r`
* `fabs(d_x) = 8*r` and `xl/2. = 5*r`
* `8*r > 5*r` so `d_x = xl - fabs(d_x) = 10*r - 8*r = 2*r`
* `2*r != 8*r` and the actual distance is `8*r` due to the boundary conditions